src/components/Session/WithAuthentication.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import AuthUserContext from './context';
+import {withFirebase} from '../Firebase';
+
+const withAuthentication = Component => {
+  class WithAuthentication extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        authUser: null
+      };
+    }
+    componentDidMount() {
+      this.listener = this.props.firebase
+        ? this.props.firebase.auth.onAuthStateChanged(authUser => {
+            authUser
+              ? this.setState({authUser})
+              : this.setState({authUser: null});
+          })
+        : null;
+    }
+    componentWillUnmount() {
+      this.listener();
```

Comment:
Kas on võimalik olukord, kus componentDidMount'is on `this.props.firebase` undefined vms, nii et `this.listener === null`? Kui jah, siis siin tekiks type error.